packages/@lwc/synthetic-shadow/src/faux-shadow/shadow-root.ts

```diff
@@ -97,24 +97,24 @@ export function isDelegatingFocus(host: HTMLElement): boolean {
     return getInternalSlot(host).delegatesFocus;
 }
 
-export function getHost(root: SyntheticShadowRootInterface): HTMLElement {
+export function getHost(root: SyntheticShadowRootInterface): Element {
     return getInternalSlot(root).host;
 }
 
-export function getShadowRoot(elm: HTMLElement): SyntheticShadowRootInterface {
+export function getShadowRoot(elm: Element): SyntheticShadowRootInterface {
     return getInternalSlot(elm).shadowRoot;
 }
 
-export function hasSyntheticShadow(elm: HTMLElement): boolean {
+// Intentionally adding Node here as possible the first argument
+// since this check is harmless for nodes as well, and it speeds up things
+// to avoid casting before calling this method in few places.
+export function hasSyntheticShadow(elm: Element | Node): boolean {
```

Comment:
How about naming this `isHostElement()`?

Reply:
good call... will do